rmw_dds_common/test/test_security.cpp

```diff
@@ -0,0 +1,112 @@
+  }
+
+  std::unordered_map<std::string, std::string> security_files;
+  ASSERT_TRUE(rmw_dds_common::get_security_files("", dir.string(), security_files));
+
+  EXPECT_EQ(security_files["IDENTITY_CA"], "./test_folder/identity_ca.cert.pem");
+  EXPECT_EQ(security_files["CERTIFICATE"], "./test_folder/cert.pem");
+  EXPECT_EQ(security_files["PRIVATE_KEY"], "./test_folder/key.pem");
+  EXPECT_EQ(security_files["PERMISSIONS_CA"], "./test_folder/permissions_ca.cert.pem");
+  EXPECT_EQ(security_files["GOVERNANCE"], "./test_folder/governance.p7s");
+  EXPECT_EQ(security_files["PERMISSIONS"], "./test_folder/permissions.p7s");
+}
+
+TEST(test_security, files_exist_with_prefix)
+{
+  rcpputils::fs::path dir = rcpputils::fs::path(is_win32 ? R"(.\test_folder)" : R"(./test_folder)");
+  (void)rcpputils::fs::remove_all(dir);
+  EXPECT_TRUE(rcpputils::fs::create_directories(dir));
+  EXPECT_TRUE(rcpputils::fs::exists(dir));
+  EXPECT_TRUE(rcpputils::fs::is_directory(dir));
+
+  std::array<std::string, 6> required_files = {
+    "identity_ca.cert.pem", "cert.pem", "key.pem",
+    "permissions_ca.cert.pem", "governance.p7s", "permissions.p7s"
+  };
+  for (const std::string & filename : required_files) {
+    rcpputils::fs::path full_path = dir / filename;
+    std::ofstream output_buffer{full_path.string()};
+    output_buffer << "test";
+    ASSERT_TRUE(rcpputils::fs::exists(full_path));
+  }
+
+  std::unordered_map<std::string, std::string> security_files;
+  ASSERT_TRUE(rmw_dds_common::get_security_files("file://", dir.string(), security_files));
+
+  EXPECT_EQ(security_files["IDENTITY_CA"], "file://./test_folder/identity_ca.cert.pem");
+  EXPECT_EQ(security_files["CERTIFICATE"], "file://./test_folder/cert.pem");
+  EXPECT_EQ(security_files["PRIVATE_KEY"], "file://./test_folder/key.pem");
+  EXPECT_EQ(security_files["PERMISSIONS_CA"], "file://./test_folder/permissions_ca.cert.pem");
+  EXPECT_EQ(security_files["GOVERNANCE"], "file://./test_folder/governance.p7s");
+  EXPECT_EQ(security_files["PERMISSIONS"], "file://./test_folder/permissions.p7s");
+}
+
+TEST(test_security, file_missing)
+{
+  rcpputils::fs::path dir = rcpputils::fs::path(is_win32 ? R"(.\test_folder)" : R"(./test_folder)");
+  (void)rcpputils::fs::remove_all(dir);
+  EXPECT_TRUE(rcpputils::fs::create_directories(dir));
+  EXPECT_TRUE(rcpputils::fs::exists(dir));
+  EXPECT_TRUE(rcpputils::fs::is_directory(dir));
+
+  std::array<std::string, 5> required_files = {
+    "identity_ca.cert.pem", "cert.pem", "key.pem",
+    "permissions_ca.cert.pem", "governance.p7s"
+  };
+  for (const std::string & filename : required_files) {
+    rcpputils::fs::path full_path = dir / filename;
+    std::ofstream output_buffer{full_path.string()};
+    output_buffer << "test";
+    ASSERT_TRUE(rcpputils::fs::exists(full_path));
+  }
+
+  std::unordered_map<std::string, std::string> security_files;
+  ASSERT_FALSE(rmw_dds_common::get_security_files("", dir.string(), security_files));
```

Comment:
@clalancette nit: documentation says `security_files` should be empty, consider checking for it.

Reply:
Good call, fixed in 4f541d9